### PR TITLE
To Use Circle CI instead of Travis CI for builds longer than 50mins

### PR DESCRIPTION
**Pull request by AnonAlpha**
To Use Circle CI instead of Travis CI for builds longer than 50mins

### CODE REVIEW

**Comment by gautamkrishnar** [comment]
@TGYZ1909 Thanks a lot for contributing. I am not planning to use circle CI exclusively on this project. Can you please revert the changes to .travis.yml and update the PR. Please add a circle-ci-readme.md that contains the instructions for setting up the project if you have time. Please use #1 to store the assets.

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+version: 2
+jobs:
+  build:
+    docker: # executor type
+      - image: buildpack-deps:trusty
+    # The secondary container is an instance of the second listed image which is run in a common network where ports exposed on the primary container are available on localhost.
+    steps:
+      - add_ssh_keys:
+          fingerprints: 'Your fingerprint'
```
Comment (gautamkrishnar) [inline .circleci/config.yml]: What is the use of this. Please explain. Is it a required item. Cant we read this from environment variable?

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      - checkout
+      - run:
+          name: Update npm
+          command: |
+            sudo apt-get install curl
+            curl -sL https://deb.nodesource.com/setup_12.x | bash -
+            sudo apt-get install -y nodejs
+      - run:
+          name: Install Dependence
```
Comment (gautamkrishnar) [inline .circleci/config.yml]: Typo in the name

File: .circleci/config.yml (hunk 1 above)
Comment (AnonAlpha) [reply .circleci/config.yml]: This is how circle ci add a ssh key to the VM, I'll update the PR

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            sudo add-apt-repository -y ppa:mc3man/trusty-media
+            sudo apt-get -y update
+            sudo apt-get install -y libimage-exiftool-perl
+            sudo apt-get -y install python-software-properties
+            sudo apt-get -y install software-properties-common
+            sudo apt-get -y install ffmpeg
+            sudo apt-get -y install graphicsmagick
+            sudo apt-get -y install gifsicle
+      - run:
+          name: THUMBUP BUILD
```
Comment (gautamkrishnar) [inline .circleci/config.yml]: Typo: should be `THUMBSUP`

File: .circleci/config.yml (hunk 1 above)
Comment (AnonAlpha) [reply .circleci/config.yml]: I'm using this product for my class, but the repo get too big and Travis CI can't build it in 50mins, So I decide to use Circle CI

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+          command: |
+            sudo npm config set unsafe-perm=true
+            sudo npm install -g thumbsup
+            thumbsup --config config.json
+      - run: # git push
+          name: push-github
+          command: |
+            sudo npm install -g --silent gh-pages@2.0.1
+            git config --global user.email "Your Email"
+            git config --global user.name "Your username"
```
Comment (gautamkrishnar) [inline .circleci/config.yml]: Read this values from environment variables

File: .circleci/config.yml (hunk 1 above)
Comment (AnonAlpha) [reply .circleci/config.yml]: I'll create a .md file
I think Circle CI also have a limit in there container So I'm going to try more CI

**Review by gautamkrishnar — CHANGES_REQUESTED** [review]
Needs changes. Please see the comments.

File: .circleci/config.yml (hunk 1 above)
Comment (gautamkrishnar) [reply .circleci/config.yml]: Thanks a lot @TGYZ1909. Good to know its limitations. Will add that to the readme soon.

File: .circleci/config.yml (hunk 2 above)
Comment (AnonAlpha) [reply .circleci/config.yml]: This is the name showing in the Circle CI console,I think there is no need to change this?

Comment (gautamkrishnar) [reply .circleci/config.yml]: This should be **Install dependencies**  right?

File: .circleci/config.yml (hunk 4 above)
Comment (AnonAlpha) [reply .circleci/config.yml]: OK I'll try to change this

File: .circleci/config.yml (hunk 2 above)
Comment (AnonAlpha) [reply .circleci/config.yml]: Oops, yes I'm a chinese junior school student so I may make mistakes, sorry for that😥

File: .circleci/config.yml (hunk 1 above)
Comment (AnonAlpha) [reply .circleci/config.yml]: And there is another problem For the same repo Travis CI need more than 50mins to build, while Circle CI only need 7mins

File: .circleci/config.yml (hunk 2 above)
Comment (gautamkrishnar) [reply .circleci/config.yml]: Its okay 👍

File: .circleci/config.yml (hunk 4 above)
Comment (AnonAlpha) [reply .circleci/config.yml]: I've just seen the docs of Circle CI ,but Circle CI don't support environment variables in commands, I'll make a tutorial with images then.

**Review by gautamkrishnar — APPROVED** [review]

**Comment by gautamkrishnar** [comment]
@TGYZ1909 thanks for the contribution. Sorry for the delay, I totally forgot about this. Merged this to a separate branch so that people can use it if they want.

**Pull request merged**